src/vmm/src/vstate/vm.rs

```diff
@@ -377,77 +447,47 @@ pub struct VmState {
 
 #[cfg(test)]
 pub(crate) mod tests {
-    #![allow(clippy::undocumented_unsafe_blocks)]
-    use std::os::unix::io::FromRawFd;
-
     use utils::vm_memory::GuestAddress;
 
     use super::*;
-    use crate::vstate::system::KvmContext;
 
     // Auxiliary function being used throughout the tests.
     pub(crate) fn setup_vm(mem_size: usize) -> (Vm, GuestMemoryMmap) {
-        let kvm = KvmContext::new().unwrap();
         let gm = utils::vm_memory::test_utils::create_anon_guest_memory(
             &[(GuestAddress(0), mem_size)],
             false,
         )
         .unwrap();
 
-        let mut vm = Vm::new(kvm.fd()).expect("Cannot create new vm");
-        assert!(vm.memory_init(&gm, kvm.max_memslots(), false).is_ok());
+        let vm = Vm::new().expect("Cannot create new vm");
+        assert!(vm.memory_init(&gm, false).is_ok());
 
         (vm, gm)
     }
 
     #[test]
     fn test_new() {
-        use std::os::unix::io::AsRawFd;
-
-        use utils::tempfile::TempFile;
-        // Testing an error case.
-        let vm =
-            Vm::new(&unsafe { Kvm::from_raw_fd(TempFile::new().unwrap().as_file().as_raw_fd()) });
-        assert!(vm.is_err());
-
         // Testing with a valid /dev/kvm descriptor.
-        let kvm = KvmContext::new().unwrap();
-        assert!(Vm::new(kvm.fd()).is_ok());
-    }
-
-    #[test]
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
-    fn test_get_supported_cpuid() {
-        let kvm = KvmContext::new().unwrap();
-        let vm = Vm::new(kvm.fd()).expect("Cannot create new vm");
-        let cpuid = kvm
-            .fd()
-            .get_supported_cpuid(KVM_MAX_CPUID_ENTRIES)
-            .expect("Cannot get supported cpuid");
-        assert_eq!(vm.supported_cpuid().as_slice(), cpuid.as_slice());
```

Comment:
It was removed because `KvmContext` does not exist anymore. Also it does not make much sense. It basically tests that 2 calls to the `get_supported_cpuid` return same value.

Reply:
ah you're right. The KVM_GET_SUPPORTED_CPUID call is included in `Vm::new()` now. Resolving this comment.